src/util/performance/PerformanceUtil.cpp

```diff
@@ -228,3 +349,404 @@ int PerformanceUtil::collectLocalPerformanceData(std::string isVMStatManager, st
+                    string slaPerfSql = "insert into graph_place_sla_performance (graph_sla_id, place_id, memory_usage) "
+                                        "values ('" + graphSlaId + "','" + placeId + "', '" + memoryUsage + "')";
+
+                    perfDb.runInsert(slaPerfSql);
+                } else {
+                    string graphPlaceSlaPerfId = slaPerfResult[0][0].second;
+                    string memoryUsageString = slaPerfResult[0][1].second;
+
+                    long currentAvgMemory = atol(memoryUsageString.c_str());
+                    long currentConsumption = atol(memoryUsage.c_str());
+
+                    long newAvgMemory = ((currentAvgMemory * iteration) + currentConsumption) / (iteration + 1);
+
+                    string slaPerfUpdate = "update graph_place_sla_performance set memory_usage = '" + to_string(newAvgMemory) +
+                            "' where id = '" + graphPlaceSlaPerfId + "';";
+
+                    perfDb.runUpdate(slaPerfUpdate);
+                }
+
+            }
+        }
+    }
+}
+
+int PerformanceUtil::collectLocalSLAResourceUtilization(std::string isVMStatManager,
+                                                        std::string isResourceAllocationRequired, std::string hostId,
+                                                        std::string placeId, std::string graphId, std::string slaCategoryId,
+                                                        int iteration, int partitionCount) {
+    StatisticCollector statisticCollector;
+    Utils utils;
+    statisticCollector.init();
+    string graphSlaId;
+
+    int memoryUsage = statisticCollector.getMemoryUsageByProcess();
+    double cpuUsage = statisticCollector.getCpuUsage();
+
+    auto executedTime = std::chrono::system_clock::now();
+    std::time_t reportTime = std::chrono::system_clock::to_time_t(executedTime);
+    std::string reportTimeString(std::ctime(&reportTime));
+    reportTimeString = utils.trim_copy(reportTimeString, " \f\n\r\t\v");
+
+    if (isVMStatManager.find("true") != std::string::npos) {
+        std::string vmLevelStatistics = statisticCollector.collectVMStatistics(isVMStatManager, isResourceAllocationRequired);
+        std::vector<std::string> strArr = Utils::split(vmLevelStatistics, ',');
+
+        string totalMemoryUsed = strArr[0];
+        string totalCPUUsage = strArr[1];
+
+        string vmPerformanceSql = "insert into host_performance_data (date_time, memory_usage, cpu_usage, idhost) values ('" + reportTimeString +
+                                  "','" + totalMemoryUsed + "','"+ totalCPUUsage +"','" + hostId + "')";
+
+        perfDb.runInsert(vmPerformanceSql);
+
+        if (isResourceAllocationRequired == "true") {
+            std::string totalMemory = strArr[2];
+            std::string totalCores = strArr[3];
+            string allocationUpdateSql = "update host set total_cpu_cores='" + totalCores + "',total_memory='" + totalMemory + "' where idhost='" + hostId + "'";
```

Comment:
This line is lengthy. Lets break into two.

Reply:
Line segmented

Reply:
Done